### ISSUE TITLE
Rule proposal: require to specify rules to be disabled in "eslint-disable" comment directives

### ISSUE THREAD

**Issue by pnevyk** [eslint/eslint#11020]
**Please describe what the rule should do:**

ESLint allows to disable itself via special comments `/* eslint-disable */`, `// eslint-disable-line` and `// eslint-disable-next-line`. These comments either disable entire ESLint for intended part of code or the user can specify what exact rules are disabled. I propose to add a rule which would require to explicitly name the rules to disable.

In my opinion these directives (especially the line variants) serve mainly for disabling specific rules which given code violates and there is a reason why it cannot (or is not needed to) be fixed. However, when the user disables ESLint for all rules, it might hide an error which was not intended to be suppressed. Therefore code style violations can occur in the source code unnecessarily. Explicit enumeration of rules also gives a notion why exactly is ESLint disabled there.

**What category of rule is this? (place an "X" next to just one item)**

[ ] Enforces code style
[x] Warns about a potential error
[ ] Suggests an alternate way of doing something
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
// eslint-disable-next-line  -> rules must be enumerated explicitly
function foo(unused) {}

var PI = 3.14; // eslint-disable-line  -> rules must be enumerated explicitly

// eslint-disable-next-line no-unused-vars  -> no warning
function foo(unused) {}
```

**Why should this rule be included in ESLint (instead of a plugin)?**

Although I am not sure if this rule is very important to ESLint users, to me it seems to be very related to ESLint core and how it behaves and is used. It's not a lint for a library or a specific environment. However I will of course accept the conclusion that it is a very minor improvement and as so it doesn't belong to the core.

**Comment by not-an-aardvark** [comment]
Hi, thanks for the proposal. Does [this rule](https://github.com/mysticatea/eslint-plugin-eslint-comments/blob/c427c1c0577fab56f2ff0919053d249938507c83/docs/rules/no-unlimited-disable.md) from `eslint-plugin-eslint-comments` cover what you have in mind?

**Comment by pnevyk** [comment]
@not-an-aardvark Hi, yes that's exactly it. I apologize, I searched in eslint issues if there was such a proposal, but I should also have to look around if there is a plugin already implementing it. I am closing this issue. Thank you for your quick response.
